lib/Makefile

```diff
@@ -24,12 +24,20 @@ FF_KNI=1
 #FF_IPFW=1
 
 ifeq ($(FF_DPDK),)
+ifeq (${shell uname -m},aarch64)
```

Comment:
Why not using `MACHINE_CPUARCH`?

Reply:
OK... then how about this:
diff --git a/lib/Makefile b/lib/Makefile
index b901b1b..ddc9b89 100644
--- a/lib/Makefile
+++ b/lib/Makefile
@@ -23,8 +23,10 @@ FF_KNI=1
 #FF_NETGRAPH=1
 #FF_IPFW=1

+include ${TOPDIR}/mk/kern.pre.mk
+
 ifeq ($(FF_DPDK),)
-ifeq ($(shell uname -m),aarch64)
+ifeq ($(MACHINE_CPUARCH),aarch64)
     FF_DPDK=${TOPDIR}/dpdk/build
 else
     $(warning ${FF_DPDK})
@@ -33,7 +35,7 @@ endif
 endif

 ifdef RTE_SDK
-ifeq (${shell uname -m},aarch64)
+ifeq ($(MACHINE_CPUARCH),aarch64)
     FF_DPDK=${RTE_SDK}/build
 else
     FF_DPDK=${RTE_SDK}/x86_64-native-linuxapp-gcc
@@ -46,8 +48,6 @@ DPDK_CFLAGS+= -DRTE_MACHINE_CPUFLAG_SSSE3 -DRTE_MACHINE_CPUFLAG_SSE4_1 -DRTE_MAC
 DPDK_CFLAGS+= -DRTE_COMPILE_TIME_CPUFLAGS=RTE_CPUFLAG_SSE,RTE_CPUFLAG_SSE2,RTE_CPUFLAG_SSE3,RTE_CPUFLAG_SSSE3,RTE_CPUFLAG_SSE4_1,RTE_CPUFLAG_SSE4_2
 DPDK_CFLAGS+= -I${FF_DPDK}/include

-include ${TOPDIR}/mk/kern.pre.mk
-
 KERNPREINCLUDES:= ${INCLUDES}
 INCLUDES= -I${OVERRIDE_INCLUDES_ROOT} ${KERNPREINCLUDES}
 INCLUDES+= -I./machine_include

Reply:
Emm, this looks good, you can commit it.
Thanks for your contribution.

Reply:
Commited. Thanks.